### PR TITLE
Added solution 5.1

### PR DESCRIPTION
**Pull request by DmGrt**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,77 @@
...
+        isElectricCar = electricCar;
+    }
+
+    @Override
+    public Car clone() {
+        Car car;
+        try {
+            car = (Car) super.clone();
+        } catch (CloneNotSupportedException e) {
+            car = new Car(this.getName(), this.getEngine().clone(),
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Car.java:72]: what about checklist? 
https://mate-academy.github.io/jv-program-fulltime/01_core/checklist/hw_05.html

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,77 @@
...
+    public void setElectricCar(boolean electricCar) {
+        isElectricCar = electricCar;
+    }
+
+    @Override
+    public Car clone() {
+        Car car;
+        try {
+            car = (Car) super.clone();
+        } catch (CloneNotSupportedException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Car.java:71]: this try catch is redundant here, just use impl with constructor without try catch block

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,92 @@
...
+        if (idleAnalysis != null) {
+            this.idleAnalysis = idleAnalysis;
+        }
+    }
+
+    @Override
+    public Engine clone() {
+        try {
+            return (Engine) super.clone();
+        } catch (CloneNotSupportedException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Engine.java:87]: same here

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,71 @@
...
+            this.engine = engine;
+        }
+    }
+
+    public String getColor() {
+        return color;
+    }
+
+    public void setColor(String color) {
+        if (color != null) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java:43]: this check is redundant

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,71 @@
...
+            this.name = name;
+        }
+    }
+
+    public Engine getEngine() {
+        return engine;
+    }
+
+    public void setEngine(Engine engine) {
+        if (engine != null) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java:33]: this check is redundant

File: src/main/java/core/basesyntax/Car.java [hunk 6]
```diff
@@ -1,5 +1,71 @@
...
+        this.type = type;
+        this.isElectricCar = isElectricCar;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        if (name != null) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java:23]: this check is redundant

File: src/main/java/core/basesyntax/Car.java [hunk 7]
```diff
@@ -1,5 +1,71 @@
...
+            this.color = color;
+        }
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        if (type != null) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java:53]: this check is redundant

File: src/main/java/core/basesyntax/Engine.java [hunk 8]
```diff
@@ -1,5 +1,89 @@
...
+        this.turboPressure = turboPressure;
+        this.idleAnalysis = idleAnalysis;
+    }
+
+    public int getRpm() {
+        return rpm;
+    }
+
+    public void setRpm(int rpm) {
+        if (rpm != 0) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Engine.java:28]: this check is redundant. Please check your code

File: src/main/java/core/basesyntax/Engine.java [hunk 9]
```diff
@@ -1,5 +1,89 @@
...
+
+    public void setIdleAnalysis(String idleAnalysis) {
+        if (idleAnalysis != null) {
+            this.idleAnalysis = idleAnalysis;
+        }
+    }
+
+    @Override
+    public Engine clone() {
+        return new Engine(getRpm(), getEngineHours(), getCoolantTemperature(),
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Engine.java:85]: Is it necessary to use getter instead of variables directly?

**Pull request closed without merging**